src/main/resources/api.yaml

```diff
@@ -207,3 +270,18 @@ definitions:
     required:
       - enrollmentPeriod
       - durationPeriod
+
+  UserAvailabilityDto:
```

Comment:
Mógłbyś rozszerzyć `UserAvailabilityDto` o required `firstName` i `secondName`? Wtedy ten obiekt byłby wykorzystany do widoku z dostępnościami userów w IO2019TEAM-63 #10 - zawierałby on do tego wymagane informacje, których użycie byłoby spójne w dokumentacji swaggera. @kjarosh dodał `UserDetailsDto` w #6 co pozwoli na stworzenie zapytania o `firstName` i `secondName` od frontu

Reply:
Jeśli dodałbym takie 2 wymagane pola, to trzeba by też przy podawaniu dostępności (PUT z frontu) je wypełniać, a to trochę chyba nie ma sensu

Ja to widzę raczej tak, że po otwarciu projektu jako admin, będzie getMembers, które zwróci wszystkich userów w projekcie, wraz z ich danymi, a getAvailabilities (w sensie pobranie tego takiego podsumowania) zwróci (userId, time1, time2, notes)

Reply:
Ogólnie to muszę dodać obiekt, który zawrę w liście i który będzie zawierał te informacje, które są w `UserAvailabilityDto` + jakieś userId. Miałem dwie opcje dostępne, obie z wadami:
- Zrobić takie `UserAvailabilityDto2`  tylko, że z `userId` (trochę bez sensu tak to duplikować)
- Dodać to czego mi brakuje do `UserAvailabilityDto` widocznego tutaj (wtedy na froncie trzeba dopisywać `userId` do requestów, które tego nie powinny wymagać)

Ale jest też trzecia, bo gdyby zmienić nazwę widocznego tutaj `UserAvailabilityDto` na coś jak `AvailabilityDto` to wtedy można dodać obiekt `UserAvailabilityDto` zawierający `AvailabilityDto` + `userId`. Co myślisz?

Reply:
No właśnie dlatego byłem za tym aby querować userów osobno, wtedy kiedy trzeba, zamiast wybierać w które miejsca dodać te dane, a w które nie i w jakiej kolejności wołać endpointy. W przypadku userów w projekcie jest ten sam problem z PUTem. Jeśli tu nie dodamy danych usera to znowu będzie niekonsystencja, no ale jak uważasz.

Reply:
Zmiana na `availabilityDto` dla mnie ma sens, zaraz poprawię.
Spróbujmy może ustalić jakąś jedną wersję:
- po wejściu na widok o którym mówi @MrMarosis jest GET na `/members' i dla każdego z nich jest pobierane Availability po userId (ewentualnie jakaś metoda, która dla listy ID zwróci listę odpowiadających Availability)
- po wejściu na widok jest GET na endpoint z availabilities, zwracający obiekty jakiejś nowej klasy, które zawierają `AvailabilityDto` razem z danymi o użytkowniku
- inne?

W tym miejscu na pewno nie dodawałbym danych usera, bo ten obiekt będzie współdzielony w dwóch różnych miejscach aplikacji (trzeba wyciągnąć wg. mnie jakąś wspólną część, a różnice zaimplementować osobno).
Trochę inaczej sprawa wyglądała z polem zawierającym czas trwania sprintu (które usunąłem), bo ten obiekt docelowo nie miał być używany nigdzie indziej.

Reply:
> Ja to widzę raczej tak, że po otwarciu projektu jako admin, będzie getMembers, które zwróci wszystkich userów w projekcie, wraz z ich danymi, a getAvailabilities (w sensie pobranie tego takiego podsumowania) zwróci (userId, time1, time2, notes)

W ramach issue 55 powstało `getUsersFromProject`

Reply:
> W tym miejscu na pewno nie dodawałbym danych usera, bo ten obiekt będzie współdzielony w dwóch różnych miejscach aplikacji (trzeba wyciągnąć wg. mnie jakąś wspólną część, a różnice zaimplementować osobno).

:+1: 

> * po wejściu na widok o którym mówi @MrMarosis jest GET na `/members' i dla każdego z nich jest pobierane Availability po userId (ewentualnie jakaś metoda, która dla listy ID zwróci listę odpowiadających Availability)
> * po wejściu na widok jest GET na endpoint z availabilities, zwracający obiekty jakiejś nowej klasy, które zawierają AvailabilityDto razem z danymi o użytkowniku

Dla mnie oba spoko - nie wiem, _jak się robi takie rzeczy_, ale nie dodawałabym tutaj danych osobowych

Reply:
Chociaż nie wiem, czy nie dodałabym tutaj `userId`/`userInProjectId`? Mamy sporo sytuacji, w których olewamy jakieś pola

Reply:
Ja w ogóle nie widzę powodu uzewnętrzniania `userInProjectId`, mamy jakiś?

Reply:
Nie mamy, dałam łamane :wink: 

Pytanie nadal aktualne: 
> Chociaż nie wiem, czy nie dodałabym tutaj `userId`~/`userInProjectId`~? Mamy sporo sytuacji, w których olewamy jakieś pola

Reply:
Zmieniłem nazwę tej klasy.
Zrobienie UserAvailability zawierającego Availability oraz userId też w sumie fajnie wygląda

Reply:
No ale `Availability` i tak jest `Availability` konkretnego usera, także osobna klasa `UserAvailability` nie jest dla mnie jakaś super intuicyjna

Reply:
No ale to nie jest konkretnego usera, tylko _jakiegoś_ usera - wiadomo czyje to jest, bo ten kto prosi/aktualizuje wie kim jest. 

W sumie można tu dodać userId, bo dużo to nie zmienia

Reply:
dodałem `userId`

Reply:
Dla mnie wszystko ok